docs/kore-syntax.md

```diff
@@ -0,0 +1,143 @@
+Syntax of the KORE language
+===========================
+
+This documents describes the syntax of KORE, as admitted by the Haskell Backend.
+
+It is meant to be an easier to access source than the
+[Semantics of K](docs/semantics-of-k.pdf)
+
+Any change in the KORE Parser infrastructure should be reflected here.
+
+Lexical Syntax
+--------------
+
+### Comments
+
+- C-style comments (non-nested)
+
+### Strings
+
+```
+StringLiteral ::= "\"" {Char}* "\""
+
+Char ::= EscapeChar || AsciiChar || PrintableChar
+
+AsciiChar ::= first 128 ASCII characters
+PrintableChar ::= printable according to the C++ iswprint function
```

Comment:
Actually their definition is pretty strange, because it depends on the current `locale` settings: I was quite surprise we do support these and I'm pretty certain that maybe we shouldn't.

Reply:
> Actually their definition is pretty strange, because it depends on the current `locale` settings

That shouldn't be the case, are you sure? The Unicode character classes are fixed. If your system is using a non-Unicode encoding, the locale settings will influence how characters are decoded from your system encoding to Unicode, but we only accept ASCII anyway. If this is really influenced by locale settings, then we need to fix the implementation.

Reply:
I based my comment on this: https://en.cppreference.com/w/c/string/wide/iswprint
and on the fact that `Data.Char.isPrint` delegates to `iswprint`: https://hackage.haskell.org/package/base-4.12.0.0/docs/src/GHC.Unicode.html#isPrint

Reply:
It does not, it delegates to `u_iswprint` which is defined somewhere in the RTS. I'm still looking it up at the moment, but I doubt it delegates to `iswprint`, which has the wrong signature on most platforms.

Reply:
`u_iswprint` is defined as this [predicate](https://gitlab.haskell.org/ghc/ghc/blob/master/libraries/base/cbits/WCsubst.c#L5054) on Unicode character classes.